dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/settings/SettingsStore.java

```diff
@@ -32,50 +33,116 @@
-            return settingsStore;
         }
+        return createDefaultSettingStore();
+    }
+
+    /**
+     * Create default settings.
+     *
+     * @return Default setting store
+     */
+    private static SettingsStore createDefaultSettingStore() {
+        LOG.info("Creating new empty saved setting store");
+        final SettingsStore settingsStore = new SettingsStore(LogSettings.DEFAULT,
+                ProcessSettings.DEFAULT, FilePropertiesSettings.DEFAULT,
+                ShowCachedJobSettings.DEFAULT);
+        // set this to true so we will write the settings after the first run
+        settingsStore.dirty = true;
+        return settingsStore;
     }
 
+    /**
+     * Save setting to setting file.
+     *
+     * @param settingsStore saved setting store
+     * @throws IOException
+     */
     public static void saveSettingsStore(final SettingsStore settingsStore) throws IOException {
         if (settingsStore.dirty) {
             LOG.info("Session store was dirty, saving...");
             if (!Files.exists(PATH.getParent())) {
                 Files.createDirectories(PATH.getParent());
             }
-            try (final OutputStream outputStream = Files.newOutputStream(PATH, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE)) {
+            try (final OutputStream outputStream = Files.newOutputStream(PATH,
+                    StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE)) {
                 JsonMapping.toJson(outputStream, settingsStore);
             }
         }
     }
 
+    /**
+     * Method to include new json entry to settings.json file if any new setting property is introduced.
+     */
+    public static void writeNewSettingsToSettingsJsonFile() {
+        try (final Stream<String> stream = Files.lines(PATH)) {
+            stream.forEach(entry -> {
+                if (!entry.contains(StringConstants.FILE_PROPERTIES_SETTING)) {
+                    appendNewFieldToSetting(entry, StringConstants.FILE_PROPERTIES_DEFAULT);
+                }
+                if (!entry.contains(StringConstants.SHOW_CACHED_JOB_SETTING)) {
+                    appendNewFieldToSetting(entry, StringConstants.SHOW_CACHED_JOB_DEFAULT);
+                }
+            });
+        } catch (final Exception e) {
+            LOG.error("Failed to save settings", e);
+        }
+    }
+
+    /**
+     * Append new setting into existing setting file
+     *
+     * @param entry          entry
+     * @param settingDefault Default setting for the file
+     */
+    private static void appendNewFieldToSetting(final String entry, final String settingDefault) {
+        final StringBuilder newFile = new StringBuilder(entry);
+        newFile.deleteCharAt(newFile.length() - 1);
+        newFile.append(settingDefault);
+        try (final BufferedWriter writer = Files.newBufferedWriter(PATH)) {
+            writer.write(newFile.toString());
+        } catch (final IOException io) {
```

Comment:
If this exception catch block is the same as the Exception catch block bellow it, delete this